tests/visco_plastic_adiabatic_heating_density.prm

```diff
@@ -0,0 +1,113 @@
+# A test with the visco_plastic material model to ensure
+# that the multicomponent_incompressible EOS module
+# uses the adiabatic temperature to compute density when
+# adiabatic heating is turned on. The formula specified
+# in this test follows the Extended Boussinesq approximation.
+# The initial (constant) temperature throughout the model
+# has the same value as the Adiabatic surface temperature.
+# As a result, the density at the top (surface) of the model
+# should  be equal to the background value specified in the 
+# material model (3300):
+#   density = background_density * ( 1 - thermal_expansivity * (temperature - reference_temperature)),
+# where the reference_temperature is the adiabatic temperature.
+# At the base of the model the density should be lower than at the surface
```

Comment:
Actually, since the test only does the 1st timestep, in which the temperature is prescribed, the density is constant.

Reply:
I think the density should vary, given that the calculated density in the first time step does take into account the adiabatic profile. 

However, I incorrectly stated the density should be lower at the surface, when in fact it should be higher (temperature < adiabatic temperature at base). I confirmed the calculated density at the base of the model (solution file) is consistent with the difference between the temperature and adiabatic temperature. 

Does this resolve the issue? I also updated the parameter documentation to make this more clear and address your comment below.

Reply:
Sorry, I was confused by the average density being reported as exactly the reference value 3300. I think that's just rounded off, as the gnuplot file does report the density increasing with depth to 3300.4. 

I only wonder about the `adiabatic density` that is reported. This is computed by querying the material model for the density using the adiabatic temperature (>= 1573) as the temperature, at a moment when the adiabatic profile is not initialized yet. Thus the `reference_temperature` is taken to be the `reference_T` (273), and the adiabatic density turns out to be something very different from the actual density, namely 3256.68. The difference with `rho_0` (43) is much larger than the change in density along the adiabat (0.4). This seems wrong to me, but I'm not sure how to fix it. Perhaps a third condition where `this->include_adiabatic_heating() && !this->get_adiabatic_conditions().is_initialized()` where the surface adiabatic temperature is taken as reference temperature (since you want to keep the `reference_T` for now)?

Reply:
@anne-glerum - good catch, that is definitely an issue! I ran the test for more than one time step and as anticipated the adiabatic density does not change. 

I'm actually a bit confused about the order of operations here, which I've tried to write out below: 

1. The adiabatic density in this case is being calculated in `adiabatic_conditions/compute_profile.cc`
2. The density in `adiabatic_conditions/compute_profile.cc` is derived from the material model, which prior to initialization of the adiabatic profile (happening concurrently) is use `reference_T` (273) rather than the adiabatic surface temperature?
3. Once the adiabatic conditions have been initialized, the adiabatic profile is not updated afterwards to account for the adiabatic surface temperature.

Is the above an accurate summary of the issue?

I agree adding the following third condition to use the surface adiabatic temperature will likely fix the issue:
`this->include_adiabatic_heating() && !this->get_adiabatic_conditions().is_initialized()`

I'll give it a shot and fingers crossed! More reason to remove `reference_Temperature` sooner rather than later.

Reply:
Adding `this->include_adiabatic_heating() && !this->get_adiabatic_conditions().is_initialized()` did the trick 👍 - the adiabatic density now looks correct.